plugins/modules/azure_rm_recoveryservicesvault.py

```diff
@@ -0,0 +1,314 @@
+                - The location of the resource.
+            returned: always
+            type: str
+            sample: "eastus"
+        name:
+            description:
+                - Name of the recovery services vault name.
+            returned: always
+            type: str
+            sample: revault_name
+        properties:
+            description:
+                - The recovery service vault properties.
+            returned: always
+            type: dict
+            sample: {
+                    "privateEndpointStateForBackup": "None",
+                    "privateEndpointStateForSiteRecovery": "None",
+                    "provisioningState": "Succeeded"
+                    }
+        sku:
+            description:
+                - The sku type of the recovery service vault.
+            returned: always
+            type: str
+            sample: Standard
+        type:
+            description:
+                - The type of the recovery service vault.
+            returned: always
+            type: str
+            sample: "Microsoft.RecoveryServices/vaults"
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_rest import GenericRestClient
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_ext import AzureRMModuleBaseExt
+import re
+import json
+import time
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+class AzureRMRecoveryServicesVault(AzureRMModuleBaseExt):
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.state = None
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.url = None
+        self.status_code = [200, 201, 202, 204]
+
+        self.body = {}
+        self.query_parameters = {}
+        self.query_parameters['api-version'] = None
+        self.header_parameters = {}
+        self.header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+
+        super(AzureRMRecoveryServicesVault, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                            supports_check_mode=True,
+                                            supports_tags=True)
+
+    def get_api_version(self):
+        return '2016-06-01' if self.state == 'present' or self.state == 'absent' else '2019-05-13'
```

Comment:
Why is the old api used when it is created, and the newer api used for deletion?

Reply:
for both, we are using 2016-06-01 version only.
the condition is 

if state is present or absent
    return 2016-06-01
else:
    return '2019-05-13'

for creation and deletion, 2016-06-01 version is provided in the official document.